### PR TITLE
Vertically align icons added to form descriptions

### PR DESCRIPTION
**Pull request by bergice**
Relates to https://github.com/silverstripe/silverstripe-userforms/issues/949

### CODE REVIEW

File: client/src/components/Form/Form.scss [hunk 1]
```diff
@@ -131,6 +131,14 @@
   &:empty {
     display: none;
   }
+
+  .icon {
+    &::before {
+      vertical-align: middle;
+      margin-right: 6px;
```
Comment (Cheddam) [inline client/src/components/Form/Form.scss:138]: What happens if the icon is used at the end of the description? 🤔

File: client/src/components/Form/Form.scss [hunk 2]
```diff
@@ -131,6 +131,14 @@
...
+      font-size: 16px;
```
Comment (Cheddam) [inline client/src/components/Form/Form.scss:139]: Is there a variable we can use here instead of hardcoding this size?

Comment (bergice) [reply client/src/components/Form/Form.scss:139]: Perhaps:

```suggestion
      font-size: $font-size-lg;
```

@sachajudd any ideas?

File: client/src/components/Form/Form.scss (hunk 1 above)
Comment (bergice) [reply client/src/components/Form/Form.scss]: No margin is shown between the text and the icon.

Some potential solutions:
* Wrap text in `<p>` tag to be able to correctly apply margin
* Treat description blurb as single element and expect people to add whitespace `&nbsp` between icons
* Add icon classes to the description element

May be out of scope for this piece of work though.

File: client/src/components/Form/Form.scss [hunk 3]
```diff
@@ -123,6 +123,8 @@
 
 .form__field-extra-label,
 .form__field-description {
+  display: flex;
```
Comment (bergice) [inline client/src/components/Form/Form.scss:126]: @Cheddam I'm confused about why this was added. It seems to mis-align the content.

Comment (Cheddam) [reply client/src/components/Form/Form.scss:126]: Can you provide an example of this misalignment? I introduced it (along with `align-items: center;`) in order to fix the alignment, not break it 😅

File: client/src/components/Form/Form.scss (hunk 2 above)
Comment (Cheddam) [reply client/src/components/Form/Form.scss:139]: I thought we'd run into trouble with using this variable (as it calculates out to `16.003px`) but I've just tested it again and it seems fine, so yep apply it 👍

File: client/src/components/Form/Form.scss (hunk 3 above)
Comment (bergice) [reply client/src/components/Form/Form.scss:126]: Without flex, align:

image

With flex, align:

image

Comment (sachajudd) [reply client/src/components/Form/Form.scss:126]: Can you check how Firefox vs Chrome behaves?

File: client/src/components/Form/Form.scss (hunk 1 above)
Comment (Cheddam) [reply client/src/components/Form/Form.scss:141]: I think putting the icon at the start of the description is likely the most common use-case, and we can improve this later, so happy to proceed for now.

File: client/src/components/Form/Form.scss (hunk 3 above)
Comment (Cheddam) [reply client/src/components/Form/Form.scss:126]: I'm really not sure how you're hitting this @bergice. Here's what it looks like for me in Chrome 83:

image

And this is Firefox 77:

image

Comment (bergice) [reply client/src/components/Form/Form.scss:126]: Okay, me neither. I just tested it again and it looks fine. 🤷

**Comment by ScopeyNZ** [comment]
Hey @bergice, in the future can you make sure that you don't create merge commits on the base branch (please rebase), and you don't self-merge un-approved work so soon after adding changes?

Thanks

**Comment by michalkleiner** [comment]
This caused a regression for field descriptions containing markup. We rely quite extensively on being able to provide useful guidance to content editors, so this is quite incovenient.

Screen Shot 2020-08-09 at 5 33 12 PM

**Comment by Cheddam** [comment]
@michalkleiner Thanks for the heads up!

**Pull request merged**